test/e2e/log_size_monitoring.go

```diff
@@ -0,0 +1,219 @@
+	workChannel  chan WorkItem
+	workers      []*LogSizeGatherer // Need at least two workers, as worker 0 is dedicated to master
+}
+
+// LogSizeGatherer is a worker which grabs a WorkItem from the channel and does assigned work
+type LogSizeGatherer struct {
+	closeChannel chan bool
+	data         *map[string]map[string]*LogsSizeData
+	lock         *sync.Mutex
+	wg           *sync.WaitGroup
+	workChannel  chan WorkItem
+}
+
+// TimestampedSize contains a size together with a time of measurement.
+type TimestampedSize struct {
+	timestamp time.Time
+	size      int
+}
+
+// LogsSizeData keeps whole timesries of TimestampedSize.
+type LogsSizeData struct {
+	sizeTimeseries []TimestampedSize
+}
+
+// WorkItem is a command for a worker that containes an IP of machine from which we want to
+// gather data and paths to all files we're interested in.
+type WorkItem struct {
+	ip    string
+	paths []string
+}
+
+// NewLogsVerifier creates a new LogsSizeVerifier which will stop when closeChannel is closed
+func NewLogsVerifier(c *client.Client, closeChannel chan bool) *LogsSizeVerifier {
+	nodes, err := NodeSSHHosts(c)
+	expectNoError(err)
+
+	data := make(map[string]map[string]*LogsSizeData)
+	lock := &sync.Mutex{}
+	wg := &sync.WaitGroup{}
+	wg.Add(workersNo)
+	workChannel := make(chan WorkItem, len(nodes)+1)
+	workers := make([]*LogSizeGatherer, workersNo)
+
+	verifier := &LogsSizeVerifier{
+		client:       c,
+		closeChannel: closeChannel,
+		data:         data,
+		lock:         lock,
+		masterIP:     getMasterHost() + ":22",
+		nodes:        nodes,
+		wg:           wg,
+		workChannel:  workChannel,
+		workers:      workers,
+	}
+	for i := 0; i < workersNo; i++ {
+		workers[i] = &LogSizeGatherer{
+			closeChannel: closeChannel,
+			data:         &verifier.data,
+			lock:         lock,
+			wg:           wg,
+			workChannel:  workChannel,
+		}
+	}
+	return verifier
+}
+
+// PrintData returns a string with formated results
+func (v *LogsSizeVerifier) PrintData() string {
+	buf := &bytes.Buffer{}
+	w := tabwriter.NewWriter(buf, 1, 0, 1, ' ', 0)
+	fmt.Fprintf(w, "host\tlog_file\taverage_rate (B/s)\tnumber_of_probes\n")
+	for k, v := range v.data {
+		fmt.Fprintf(w, "%v\t\t\t\n", k)
+		for path, data := range v {
+			last := data.sizeTimeseries[len(data.sizeTimeseries)-1]
+			first := data.sizeTimeseries[0]
+			rate := (last.size - first.size) / int(last.timestamp.Sub(first.timestamp)/time.Second)
+			fmt.Fprintf(w, "\t%v\t%v\t%v\n", path, rate, len(data.sizeTimeseries))
+		}
+	}
+	w.Flush()
+	return buf.String()
+}
+
+// Run starts log size gathering. It starts a gorouting for every worker and then blocks until closeChannel is closed
+func (v *LogsSizeVerifier) Run() {
+	v.workChannel <- WorkItem{
+		ip:    v.masterIP,
+		paths: masterLogsToCheck,
+	}
+	for _, node := range v.nodes {
+		v.workChannel <- WorkItem{
+			ip:    node,
+			paths: nodeLogsToCheck,
+		}
+	}
+	for _, worker := range v.workers {
+		go func(w *LogSizeGatherer) {
+			shouldWork := true
```

Comment:
I think the loop logic should be hidden in the Worker.

What you should do here is just:

    go worker.Run()